index.bs

```diff
@@ -1784,6 +1788,59 @@ endpoint is verified to be a WebTransport endpoint; thus, the Web application
 cannot distinguish between a non-existing endpoint and the endpoint that is not
 willing to accept connections from the Web.
 
+## Authentication using Certificate Hashes {#certificate-hashes}
+
+Normally, a user agent authenticates a TLS connection between itself and a
+remote endpoint by verifying the validity of the TLS server certificate
+provided against the server name in the URL.  This is accomplished by chaining
+server certificates to one of the trust anchors maintained by the browser; the
+trust anchors in question are responisble for authenticating the server names
+in the certificates.  We will refer to this system as Web PKI.
+
+This API provides web applications with a capability to connect to a remote
+network endpoint authenticated by a specific server certificate, rather than
+its server name.  This mechanism enables connections to endpoints for which
+getting long-term certificates can be challenging, including hosts that are
+ephemeral in nature (e.g. short-lived virtual machines), or that are not
+publicly routable.  Since this mechanism substitutes Web PKI-based
+authentication for an individual connection, we need to compare the security
+properties of both.
+
+A remote server will be able to successfully perform a TLS handshake only if it
+posesses the private key corresponding to the public key of the certificate
+specified.  The API identifies the certificates using their hashes.  That is
+only secure as long as the cryptographic hash function used has second-preimage
+resistance.  The only function defined in this document is SHA-256; the API
+provides a way to introduce new hash functions through allowing multiple
+algorithm-hash pairs to be specified.
+
+It is important to note that Web PKI provides additional security
+mechanisms in addition to simply establishing a chain of trust for a server
+name.  One of them is handling certificate revocation.  In cases where the
+certificate used is ephemeral, such mechanism is not necessary.  In other
+cases, the Web application has to consider the mechanism by which the
+certificate hashes are provisioned; for instance, if the hash is provided as a
+cached HTTP resource, the cache needs to be invalidated if the corresponding
+certificate has been rotated due to compromise.
+
+Web PKI enforces an expiry period requirement on the certificates.  This
+requirement limits the scope of potential key compromise; it also forces server
+operators to design systems that support and actively perform key rotation.
+For this reason, WebTransport imposes a similar expiry requirement; as the
+certificates are expected to be ephemeral or short-lived, the expiry period is
+limited to two weeks.
```

Comment:
Maybe connect this to the revocation thing above; this also bounds the consequences of cache mishaps from the previous paragraph.

Reply:
+1.